### ISSUE TITLE
Randomness

### ISSUE THREAD

**Issue by snarb** [kwea123/ngp_pl#85]
Hello. Thanks.
I am observing that training has significant variance because of some random factors. I have made several runs with exactly same training settings. But got different depth and quality of results from run to run. Like PSNR from  28.3 to 29.2. Visually in one case there is dark blurry spot artifacts and in other - no. Depth estimation is also significantly different from first training steps to the finish. No matter how long to train. What could be the reason? Weights initialization? Looks like in original instant ngp, there is no such randomness.

**Comment by kwea123** [comment]
> Depth estimation is also significantly different from first training steps to the finish. No matter how long to train.
I do not understand this, as training proceeds the result is naturally different.

> What could be the reason? Weights initialization?
Weight initialization is one thing, and random batch of training rays, randomness in ray marching, all of these causes randomness.

>  Looks like in original instant ngp, there is no such randomness.
I didn't pay attention to this, but I believe they also introduce randomness like above, you can search for the keyword `rng` in their code, there are a lot.

**Comment by snarb** [comment]
> I do not understand this, as training proceeds the result is naturally different.
> 
> >
I mean that I think if we are overfitting to the same set of fixed views with the same camera parameters if the training go and go on and learning rate decreases we should in expectation converge to the +- same scene representation.


Here are examples on 10 "bad" runs ![bad](https://user-images.githubusercontent.com/4518007/195337444-4215a1e4-fbd7-417b-969b-bae5de6368cd.png)   with visible black fog artifacts  I have ~1 good run ![good](https://user-images.githubusercontent.com/4518007/195337605-40aaa064-d453-4b3e-a8b1-48b86247e00a.png)
 without artifacts  . And trying to find the reason behind it.  Now playing with distortion_loss_w. Looks like it is working more stable. But still often have such kind of depth estimated after first 200 steps , then I know training will diverge. 
![depth](https://user-images.githubusercontent.com/4518007/195338779-09a53eb6-8ca6-4b33-95fb-5c0ea3fefa7e.png)
Looks like with carefully selected weight it is much better and stable than default loss for me. 

Will also try to set seed for torch random generator. And increase batch size.

**Comment by kwea123** [comment]
For real scene, the `scale` factor is also quite important in my experiments, without prior knowledge, it is better to set it to a higher value e.g. 16, and if the training goes well, try to decrease it in the next experiment.

**Comment by snarb** [comment]
Scale adjustment was my first step. In the Nvidia repo I have just set aab scale 16 and it produced good test view from the 1 try.  Here I have made a grid search of the scale. 28 worked the best. But than found that random factor plays a lot and created this issue. Maybe one of the reasons of higher randomness only 20 views in the canera rig(this is a Dynerf dataset). And that is considered not much for nerf
